### DISCUSSION TITLE
Any thoughts to have consistent features? (ie When doing a transfer the From & To boxes open on Enter for selection plus extra)

### DISCUSSION THREAD

**Discussion by SJK55** [moneymanagerex/moneymanagerex#4961] [Ideas]
I noticed in the latest release some improvements which aren't bad at all.
I tried a Transfer and thought about the 'hidden' feature of Enter on a box and was disappointed when the From and To boxes didn't respond the same way as the Category box does. 
Can it be considered.

Also Type and Status. Any way these could be in Alphabetical order? 

Additional. Hover time for message to appear in description too long!
![Transaction - Type order](https://user-images.githubusercontent.com/14268231/185820083-ed2d115e-ed9c-4b27-9581-9059ce48847d.jpg)
![Transaction - Status order](https://user-images.githubusercontent.com/14268231/185820088-fad9066f-327d-4dae-91c9-0f67017ee228.jpg)
![Hover Text too long](https://user-images.githubusercontent.com/14268231/185820091-5b932f46-8c6b-4b92-b3af-05cde4625c75.jpg)

**Comment by tactilis** [comment]
@SJK55
> I tried a Transfer and thought about the 'hidden' feature of Enter on a box and was disappointed when the From and To boxes didn't respond the same way as the Category box does.

There is no equivalent of the Organize Categories dialog (or similarly, the Organize Payees dialog) for Accounts that would be displayed by pressing Enter in the From/To comboboxes.

What benefit would such a dialog have compared to just filtering the accounts by typing a few characters, or just pressing the down arrow key, or clicking the down arrow in the field? Typically people have far fewer accounts than they do Categories (or Payees).

> Type and Status. Any way these could be in Alphabetical order?

The list of transaction Types is ordered the way it is because in general (but I acknowledge, not always) bank accounts have more Withdrawals than they do Deposits or Transfers. Therefore the dialog shows the types in the most likely order of use.

Similarly with Status. Transactions typically start off as Unreconciled and then become Reconciled. More rarely would you want to set them to Void, Follow-up, or Duplicate. If you prefer them to be Reconciled when you create them, you should can set the default in Options | Others.

It's worth exploring all of the New Transaction defaults because that can save a lot of time if you set them appropriately:
 
![image](https://user-images.githubusercontent.com/4060092/185865538-0bf83876-02b5-4f3f-b116-7f4609b11b89.png)
 
 
> Hover time for message to appear in description too long!

I wouldn't want it to be any quicker:

https://user-images.githubusercontent.com/4060092/185867725-f35a8cfc-5bbe-41ad-b1d6-c377490201ce.mp4

How long does it take on your system?

**Reply by SJK55** [reply]
Martin, mine like yours (hover time) is instant off and on, when and only when, you move the mouse focus in and out of the Notes box.
Try instead leaving the mouse in it, and depending where you start typing the text is hidden by the 'help text' (not always) and takes a while to disappear.
That's my problem. I never noticed it before the last few releases being 'in the way' when you typed information in there. Maybe the floating text could be by default lower in the box than near the text being typed?

![Floating Text](https://user-images.githubusercontent.com/14268231/185875644-97d6479f-7416-423c-8cc8-74c5b3544948.jpg)


In regards "Typically people have far fewer accounts than they do Categories (or Payees)." I for some reason have 50! (excessive?) Plus this box doesn't have a 'triangle  pull out feature as some boxes do.

Regarding the New Transactions I set the Options 'Others' to Default Date 'Last used' but when doing a duplicate of an older transaction it puts not last used but today's date. I recall a many versions ago it used to keep the date of the duplicated transaction. Not so now. In fact on or off today's date is placed in the date field every time. (I do get the prompt that Options has been updated whichever one I use)

**Reply by vomikan** [reply]
![image](https://user-images.githubusercontent.com/6836805/185880471-c9b005fc-a988-43a2-8ceb-b5d2b7baae4a.png)

**Reply by tactilis** [reply]
@SJK55
> Try instead leaving the mouse in it, and depending where you start typing the text is hidden by the 'help text' (not always) and takes a while to disappear. That's my problem. I never noticed it before the last few releases being 'in the way' when you typed information in there. Maybe the floating text could be by default lower in the box than near the text being typed?

The tooltip appears where the mouse pointer is when you move it into the field. So move rapidly to the lower part of the Notes field and click there. The text entry cursor will still be at the top of the field but the tooltip will be out of the way.

Or, as @vomikan suggests, turn off the tooltips. You don't need them once you have a good understanding of what each field in MMEX does.

> Regarding the New Transactions I set the Options 'Others' to Default Date 'Last used' but when doing a duplicate of an older transaction it puts not last used but today's date. I recall a many versions ago it used to keep the date of the duplicated transaction. Not so now. In fact on or off today's date is placed in the date field every time.

It's worth spending some time studying the MMEX Options. You problem can be solved as shown here:
 
![image](https://user-images.githubusercontent.com/4060092/185884242-878217e6-1fea-4962-b15d-1e53698bef8e.png)

**Reply by tactilis** [reply]
@SJK55
> "Typically people have far fewer accounts than they do Categories (or Payees)." I for some reason have 50! (excessive?)

Are your 50 accounts all active?

If not, then you should mark the inactive ones as Closed, which will remove them from the dropdown accounts list.

![image](https://user-images.githubusercontent.com/4060092/185886720-78af29e2-dafd-40af-bd9a-1c93b4ba858a.png)

**Reply by SJK55** [reply]
Thank you all, excellent suggestions!
I will implement them.

However once an account is setup say Checking, which ends up in "Bank Accounts" There seems no way to change it to a "Cash Account"
as the combo box is locked. This might be why I have 50 accounts!

![Edit Account](https://user-images.githubusercontent.com/14268231/186120552-fba52653-3d3a-40de-9fbe-37fdb131e268.jpg)

**Reply by SJK55** [reply]
PS I'm just after one more ask.
When doing transactions 'New' why not have the search function built directly into the combo box for smart entry? Seems a lot of extra work to find category or payee you misspell or forget and you go to the bottom of the new box to search. Just wondering?

**Comment by tactilis** [comment]
@SJK55 
> However once an account is setup say Checking, which ends up in "Bank Accounts" There seems no way to change it to a "Cash Account" as the combo box is locked.

May I gently suggest that you explore the menus a little more.

You can do:  `Accounts | Reallocate Account... | select account | select new account type`
 
> When doing transactions 'New' why not have the search function built directly into the combo box for smart entry? Seems a lot of extra work to find category or payee you misspell or forget and you go to the bottom of the new box to search.

But you can do this already. Just start typing the name of the payee or the category in the combobox and the list will be filtered:

![image](https://user-images.githubusercontent.com/4060092/186128164-497649df-67d5-453a-9f45-bbd7b2bb0b94.png)
 
At the moment, the search text is 'anchored' to the start of the payee/category. There _may_ be some improvements to this in the future to allow the entered text to be searched anywhere in the payee or category:subcategory name.

**Comment by tactilis** [comment]
@SJK55
> This might be why I have 50 accounts!

Only you can answer that.

But in general, you should create accounts in MMEX to model your real-world accounts (bank/checking, credit card, loan accounts, etc).

If you have 50 accounts in MMEX, then I'd suggest that you may be doing something less than optimal.

**Reply by SJK55** [reply]
All good. I have been a user for a while but with changes and updates obviously some things tripped me up. Thank you for all your patience.
The suggestion "But you can do this already. Just start typing the name of the payee or the category in the combo box and the list will be filtered:" only works if you know the tree. ie if not in a main heading this 'filter' does not work! That is where the fantastic search box in both Categories and Payee is 'magic' (eg Pet:Food - Dog - Chicken Stick. I can't type Dog as it won't pick it up, you need to drill down from Pet-Food to see the list.) However in the search box I type Dog, multiple tree entries appear which I can then open to find what I am actually looking for) This is what needs to be in the Payee and Category comb box as an option? This would then match the power of the search that the android version had.

One more thing, opening time of the app (I run windows 11) was blinding quick previously. Now it 'pauses' before data entry or mouse clicks can work. I noticed this in possibly the last 3 or so updates. Any reason for this?

**Comment by tactilis** [comment]
@SJK55
> eg Pet:Food - Dog - Chicken Stick. I can't type Dog as it won't pick it up, you need to drill down from Pet-Food to see the list.

Yes, that is the current behaviour.

As I mentioned above:

_"At the moment, the search text is **'anchored' to the start** of the payee/category. There _may_ be some improvements to this in the future to allow the entered text to be **searched anywhere in the payee or category:subcategory name**."_

It would need some usability tests to work out the best way to implement this. For example, at the moment, if you want to filter on a category of Education, you type 'e' and you get a short list: `Education/:Books/:Others/:Tuition`, but if we were to search for 'e' anywhere in category:subcategory, then the initial list would be over 50 items long.

This will be looked at eventually. There are other things to fix first.

**Comment by tactilis** [comment]
@SJK55
> One more thing, opening time of the app (I run windows 11) was blinding quick previously. Now it 'pauses' before data entry or mouse clicks can work. I noticed this in possibly the last 3 or so updates. Any reason for this?

Do you mean a _single_ pause after the main screen first opens before any typing or mouse clicks work?

Or, do you mean that there are pauses throughout your use of the application when typing or mouse clicking?

How long is the pause?

**Reply by vomikan** [reply]
What you are describing does not seem to be the case https://github.com/moneymanagerex/moneymanagerex/issues/4625.
After correcting this case, if you have a large payee list, there may be a delay at the moment when the cursor becomes in the Payee window.

**Comment by SJK55** [comment]
No I'm only talking about initial load time after its on screen. 1-2 seconds elapses before I can start using it. 
Like I said I didn't notice this in earlier versions. As soon as it was up and running you could start moving the window or click into the accounts. It's only noticeable as I'm sure it didn't used to be like that. Other than that it's not a app killer issue.

**Comment by tactilis** [comment]
@SJK55 
Do you show the Application Start-up window, or do you open straight into the MMEX main window?

![image](https://user-images.githubusercontent.com/4060092/186858225-2126d139-4788-4963-8652-bde559895532.png)

**Reply by SJK55** [reply]
I have that option un-ticked so it loads straight into the database.
As a test I uninstalled MMEX 1.5.19 and reinstalled version 1.3.5 (64 Bit version). Seems the delay was there too. It must be my imagination?

**Reply by SJK55** [reply]
Something that caught my attention when I used Revo-uninstaller to uninstall 1.5.19 is that it lists the type as 32-bit. Even though the installer file is named as mmex-1.5.19-win64.exe.
After my load time testing, I went to the MMEX website to fetch a fresh copy and it overwrote the one I had downloaded earlier. I re-installed and check Revo again out of curiosity. It clearly shows it listed again as 32-bit. Is this a software untruth or is the file labelled 64 bit indeed a 32-bit version?

PS I checked task manager and it's details for mmex.exe shows x64.
Not sure why Revo lists it as 32 bit?


![32 Bit maybe listed as 64 bit](https://user-images.githubusercontent.com/14268231/187100327-d7637c54-1a49-4ed6-a6e6-3dae1bbd332a.jpg)
![Task Manager 64 bit okay](https://user-images.githubusercontent.com/14268231/187100556-e937b9ac-56ba-4133-a6de-9f9aa0bca705.jpg)

**Comment by tactilis** [comment]
>  or is the file labelled 64 bit indeed a 32-bit version?

You have answered that question yourself:
> PS I checked task manager and it's details for mmex.exe shows x64.

But just to confirm...

![image](https://user-images.githubusercontent.com/4060092/187149955-3792f5f6-2983-49f9-83e4-433c5b56eecf.png)

![image](https://user-images.githubusercontent.com/4060092/187149286-22c0d23a-9964-49d9-b83d-1209d1f9ec69.png)

![image](https://user-images.githubusercontent.com/4060092/187148834-d264932e-f28b-41ba-9bdc-6eb46f11ad3e.png)

**Comment by tactilis** [comment]
@SJK55 Could you close this issue please?

It has drifted from:

_"Any thoughts to have consistent features? (ie When doing a transfer the From & To boxes open on Enter for selection plus extra)"_, which was your question about Categories

to:

- Tooltips
- Handling your 50 bank accounts
- Pauses after application start before data entry is possible
- Whether MMEX 64 bit is actually 32 bit

It's helpful to others who may be searching for information to try to keep to one topic per issue.

**Reply by SJK55** [reply]
Okay but how do I close it? I can't see any obvious control to do that.

**Comment by tactilis** [comment]
> Okay but how do I close it? I can't see any obvious control to do that.

![image](https://user-images.githubusercontent.com/4060092/187373529-5d483f29-536d-4cf7-9bba-3b2ad42bbed8.png)

**Comment by SJK55** [comment]
tactillis, I do not appear to have that feature on the screen I have access to. I even tried edit on the web page. How is it you see it and I do not?

I have attached what I see as the next comment box.
![Can't close](https://user-images.githubusercontent.com/14268231/187560073-9d13a16b-5084-49bf-aee3-11cc179ccc02.jpg)

**Comment by tactilis** [comment]
> I do not appear to have that feature on the screen I have access to

Sorry. My error.

I see now, that you opened this as a conversation in Ideas

![image](https://user-images.githubusercontent.com/4060092/187605402-9770b58c-5771-450c-8cef-5a6548d33510.png)

It's Bug reports that are able to be closed by the original  poster (and admins).

Nothing further for you to do. 

But the general advice of: 

> It's helpful to others who may be searching for information to try to keep to one topic per issue.

is still relevant for future discussions.

**Comment by SJK55** [comment]
All good. Thank you for your patience.
